internal/pkg/cli/env_init_test.go

```diff
@@ -380,10 +395,30 @@ func TestInitEnvOpts_Ask(t *testing.T) {
 						Get(envInitNamePrompt, envInitNameHelpPrompt, gomock.Any(), gomock.Any()).
 						Return("test", nil),
 					m.store.EXPECT().GetEnvironment(mockApp, mockEnv).Return(nil, nil),
+					m.envLister.EXPECT().ListEnvironments().Return([]string{}, nil),
 				)
 			},
 			wantedError: errors.New("environment test already exists"),
 		},
+		"should skip error if environment already exists in workspace": {
+			inAppName: mockApp,
+			inProfile: mockProfile,
+			inDefault: true,
+			setupMocks: func(m initEnvMocks) {
+				gomock.InOrder(
+					m.prompt.EXPECT().
+						Get(envInitNamePrompt, envInitNameHelpPrompt, gomock.Any(), gomock.Any()).
+						Return("test", nil),
+					m.store.EXPECT().GetEnvironment(mockApp, mockEnv).Return(nil, nil),
+					m.envLister.EXPECT().ListEnvironments().Return([]string{mockEnv}, nil),
+					m.sessProvider.EXPECT().FromProfile(mockProfile).Return(&session.Session{
+						Config: &aws.Config{
+							Region: aws.String("us-west-2"),
+						},
+					}, nil),
```

Comment:
nit nit nit nit nit: just to signal that we don't particularly care about `FromProfile` in this specific unit test ⬇️ 

```suggestion
	m.sessProvider.EXPECT().FromProfile(mockProfile).Return(&session.Session{
		Config: &aws.Config{
			Region: aws.String("us-west-2"),
		},
	}, nil).AnyTimes(),
```